<launch>

    <!-- Robot interface -->
    <node pkg="simple_robot_driving" type="publish_captured_data.py" name="captured_data_publisher" output="screen" />

    <!-- tf laser_frame to base_link transform -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher" output="screen" args="0 0 0 0 0 0 base_link laser_frame" />
        <!--<param name="x" value="0"/>
        <param name="y" value="0"/>
        <param name="z" value="0"/>
        <param name="yaw" value="0"/>
        <param name="pitch" value="0"/>
        <param name="roll" value="0"/>
        <param name="frame_id" value="base_link"/>
        <param name="child_frame_id" value="laser_frame"/>
    </node>-->

    <!-- Gmapping -->
    <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen">
        <rosparam command="load" file="$(find simple_robot_driving)/config/slam/gmapping_params.yaml" />
    </node>

    <!-- Move base -->
    <!--
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find simple_robot_driving)/config/navigation/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find simple_robot_driving)/config/navigation/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find simple_robot_driving)/config/navigation/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find simple_robot_driving)/config/navigation/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find simple_robot_driving)/config/navigation/base_local_planner_params.yaml" command="load" />
    </node>
    -->

    <!-- Rviz -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find simple_robot_driving)/rviz/robot_configuration.rviz" />

</launch>